includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -617,7 +617,14 @@ public function register_scripts() {
 
 		wp_set_script_translations( 'WCPAY_CHECKOUT', 'woocommerce-payments' );
 
-		if ( ! WC()->cart->needs_payment() && is_checkout() && ! has_block( 'woocommerce/checkout' ) ) {
+		// load the scripts even if cart does not need payment.
+		if (
+			isset( WC()->cart ) &&
+			! WC()->cart->is_empty() &&
```

Comment:
I am not 100% sure, I saw this check used in conjuction with `needs_payment` at other places.

Reply:
I am seeing [this place](https://github.com/Automattic/woocommerce-payments/blob/d210279f3db18865ab6881f2e89fb5144c2d3b01/includes/class-wc-payment-gateway-wcpay.php#L659-L673) in one of the changed files in this PR.

I think this would help in the case: 

```php
WC()->cart->is_empty() === true && 
WC()->cart->needs_payment() === false
```

But in the case cart is empty, I think the changing code is not called as it would be terminated earlier during the request. 

Anyway, keeping this is OK. I am marking this comment as resolved.